src/source/Ice/IceAgent.c

```diff
@@ -376,6 +376,9 @@ STATUS iceAgentAddRemoteCandidate(PIceAgent pIceAgent, PCHAR pIceCandidateString
         curr = next + 1;
     }
 
+    CHK(foundPort, STATUS_ICE_CANDIDATE_STRING_MISSING_PORT);
```

Comment:
These are good checks but we might need some unit tests around them.

Most importantly, there is a bad logic in LN 368-369. I am not sure why we set foundPort to TRUE and immediately check what we set. This needs to be properly fixed - from the look of it it seems that we need to remove LN 369.

Could you please think about how to add unit tests for these?

Reply:
Indeed, fixed it. Forgot to remove it  after modifying the code for the unit test. Speaking of the unit test, I can do it after creating those pr. For now, add the example I mention may be enough. What do you think?